celery_pubsub/pubsub.py

```diff
@@ -1,61 +1,37 @@
+import re
 import celery
 
-__all__ = ['publish', 'publish_now', 'subscribe', 'unsubscribe', ]
-
 
 class PubSubManager(object):
+
     def __init__(self):
-        super(PubSubManager, self).__init__()
-        self.subscribed = []
         self.jobs = {}
 
     def publish(self, topic, *args, **kwargs):
-        result = self.get_jobs(topic).delay(*args, **kwargs)
-        return result
+        group = []
+        for (regex, task) in self.jobs.values():
+            if regex.match(topic):
+                group.append(task)
+        out = celery.group(group)
+        return out
 
     def publish_now(self, topic, *args, **kwargs):
-        result = self.get_jobs(topic).apply(args=args, kwargs=kwargs)
+        # REVIEW: useless shortcut?
```

Comment:
Also what happens when `topic` can also be a `kwargs`?

Reply:
Interesting point. I guess this could be refactored to 
```
def publish_now(self, *args, **kwargs):
    result = self.get_jobs(args[0]).apply(args=args[1:], kwargs=kwargs)

Reply:
After giving it more thought, I will no be able to do this in `0.2.x` as it's not backward compatible. This is not the function that needs to be changed but the exposed one is.

If some of the users call it like 
```python
celery_pubsub.publish('my.topic', x=42)
celery_pubsub.publish(topic='my.topic', x=42)
```
these two currently resolve into the same call.

But if I do the change, the second will no longer work, potentially breaking things in production for some of the users.

Reply:
> it's not backward compatible.

I agree. 

Mind the fact I propose is to drop completly that method because:

```python
celery_pubsub.publish('newspaper.knowledge-base').delay('https://hyper.dev')
```

and 

```python
celery_pubsub.publish('newspaper.knowledge-base').apply(args=('http://hyper.dev`,))
```

Is much nicer than:

```python
celery_pubsub.publish('newspaper.knowledge-base', 'https://hyper.dev')
```

and 

```python
celery_pubsub.publish_now('newspaper.knowledge-base', 'http://hyper.dev`)
```

The API would be more powerful. Also people with existing knowledge of Celery, will be able to re-use their knowledge.  Conversly, the approach that is currently taken, hides some detail about the underlying celery API is wishful thinking: the user still need to poke at celery documentation at least to declare the tasks.

Reply:
I fail to see how this API would be more powerful.
Moreover, this new `publish` function would not actually publish... 😐 
And even less backward compatible. 
I'll reject this proposition but thank you for submitting it :-).